dev_tools/bash_scripts_test.py

```diff
@@ -24,13 +23,22 @@ def only_on_posix(func):
     return func
 
 
-def run(*, script_file: str, arg: str ='', setup: str = ''
-        ) -> shell_tools.CommandOutput:
+def run(
+        *,
+        script_file: str,
+        tmpdir_factory,
+        arg: str = '',
+        setup: str = '',
+) -> shell_tools.CommandOutput:
     """Invokes the given script within a temporary test environment."""
 
     with open(script_file) as f:
         script_lines = f.readlines()
 
+    # Create a unique temporary directory
+    dir_path = tmpdir_factory.mktemp('tmp', numbered=True)
+    file_path = f'{dir_path}/test-script'
```

Comment:
Don't hard code separator, file extension.

Reply:
Are you referring to `tmp`, the name of the temporary directory? Do you want it as an argument to the function instead?

Reply:
I just mean use `os.path.join` and call the file something like "test-script.sh".

Reply:
Ok, done.

Reply:
I didn't realize that "test-script" was already named.  Either way is fine.

Reply:
I agree that it makes sense to give the file its proper extension.